### DISCUSSION TITLE
Where to find psycopg2.extensions.encrypt_password in psycopg3 ?

### DISCUSSION THREAD

**Discussion by khushboovashi** [psycopg/psycopg#361] [Q&A] [answered]
We have used psycopg2.extensions.encrypt_password in pgAdmin 4. I am trying to find this method in psycopg3, but couldn't. Can someone please guide me?

**Comment by dlax** [comment, marked as answer]
There's a (low level) `encrypt_password()` method on the `pgconn` attribute of the connection:
```python
conn = psycopg.connect()
encoding = conn.info.encoding
encrypted = conn.pgconn.encrypt_password(password.encode(encoding), rolename.encode(encoding))
```

**Reply by dvarrazzo** [reply]
It must be documented, but it is part of the public interface 👍
